### PR TITLE
using a brute force approach to put all input to the discriminator

### PR DESCRIPTION
**Pull request by sunnyanthony**

### CODE REVIEW

**Review by asif256000 — COMMENTED** [review]
I am not sure if we need to use images in the discriminator. According to prof, we can just take a look at bboxes in ground truth and prediction for each object, pass all of those lists together to discriminator and let it find out if there is any error or not.

**Comment by asif256000** [comment]
> > I am not sure if we need to use images in the discriminator. According to prof, we can just take a look at bboxes in ground truth and prediction for each object, pass all of those lists together to discriminator and let it find out if there is any error or not.
> 
> In that case, the discriminator will not discriminate anything. It will be a learnable loss function instead. So, the input of discriminator will be both prediction and ground truth right?

That is what I understood from our discussion with the professor. I will let @stevend-15 confirm my understanding.

**Comment by stevend-15** [comment]
> > > I am not sure if we need to use images in the discriminator. According to prof, we can just take a look at bboxes in ground truth and prediction for each object, pass all of those lists together to discriminator and let it find out if there is any error or not.
> > 
> > 
> > In that case, the discriminator will not discriminate anything. It will be a learnable loss function instead. So, the input of discriminator will be both prediction and ground truth right?
> 
> That is what I understood from our discussion with the professor. I will let @stevend-15 confirm my understanding.

yes, that is correct from my understanding as well. The Discriminator will _**NOT**_ take images as inputs. It will receive the predictions and ground truths. From there, it should make its own predictions about whether the predictions are correct and then whether the ground truths are correct (i.e. something like `gt_output = D(gt)` and `pred_output = D(preds)`. 

I also think it is safe to completely scrap anything related to a Generator. We are only using a Discriminator.

**Comment by stevend-15** [comment]
3 concerns regarding the most recent commits: 

1) The naming scheme is pretty confusing. You take the original parameters of the `forward` method (`outputs` and `batch`) and rename them to `d_input` and `input` throughout the method. Please just keep them as their original name. 
2) A general question: what is a `stack`? We're trying to determine the meaning of the loop for `opt.num_stacks`. We understand that the original code uses the same loop, but we still do not know what a `stack` is. If you happen to know, please share. 
3) You call `d_total.backward()` and `self.D_opt.step()`. on lines 94 and 95 of `mot.py`. Shouldn't that be the very last thing that you do in the function before returning the loss values?

File: src/gen.py
Comment (stevend-15) [inline src/gen.py:1]: Class names should always be pascal case (i.e. `ThisIsATestClass`)

File: src/discriminator.py
Comment (stevend-15) [inline src/discriminator.py:1]: We should clean up the Discriminators

**Pull request merged**